test/distributed/test_c10d_functional_native.py

```diff
@@ -0,0 +1,189 @@
+# Owner(s): ["module: c10d"]
+from typing import List
+
+import torch
+import torch.distributed as dist
+from torch.testing._internal.common_distributed import (
+    MultiProcessTestCase,
+    requires_nccl,
+    skip_if_lt_x_gpu,
+)
+from torch.testing._internal.common_utils import run_tests
+
+
+if not dist.is_available():
+    print("distributed package not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+
+@requires_nccl()
+class C10DFunctionalNativeTest(MultiProcessTestCase):
```

Comment:
do we also need to test these in MultithreadedTestCase to make sure things work in multithreaded context?